Branch1

MarekKluk opened the pull request:
Added few features, described in the commits

MarekKluk committed:
Fixed white field around tortoise issue - changed Tortoise animal to Scarab

MarekKluk committed:
Fixed a bug where the organism would not disappear from the board after being killed.
Fixed an issue where the game would not continue on its own after a player was killed

MarekKluk committed:
A function has been added that rotates an organism with every move, depending on the direction

MarekKluk committed:
the map was centered horizontally and vertically in relation to the page
Removed unnecessary code in CSS

MarekKluk committed:
Added a splash screen with the description of the game.
Fixed an issue where the wasp was not rotating properly

MarekKluk committed:
Added buttons on the right side of the board, imitating a numeric board to control a character.
Fixed an issue where the character did not turn correctly to the right.

mwanago commented:
You didn't use prettier on the codebase. Please run `npm run prettier`.

mwanago commented on src/Classes/Board.js:
```diff
@@ -16,11 +16,50 @@ import Mandrake from "./plants/Mandrake";
 import Player from "./Animal/Player";
 let map = document.querySelector('.map');
```
Instead of declaring `map` as a global variable, make it a property of the `Board` class.

mwanago commented on src/Classes/Board.js:
```diff
@@ -16,11 +16,50 @@ import Mandrake from "./plants/Mandrake";
     player = null;
     constructor() {
         this.initialize();
+        this.upLeftButton = document.querySelector(".up-left-button");
```
There is quite a lot of logic here. We could fit in a separate `Controls` class.
For example, it could accept the `this.movePlayer` as the `onMove` argument.

```
class Controls {
  constructor(onMove, player) {
    // ...
  }
}
```

```
class Board {
  controls = new Controls(this.movePlayer, this.player)
  // ...
}
```

mwanago commented on src/Classes/Board.js:
```diff
@@ -122,9 +161,39 @@ export default class Board {
+        const icon = organism.icon;
+        const howXChanged = currentX - newX;
+        const howYChanged = currentY - newY;
+        if (howXChanged === -1 && howYChanged === 0){
```
Having many `if...else` statements doesn't look very good. We could improve it by creating a separate function.

```
getOrganismRotation(howXChanged, howYChanged) {
  if (howXChanged === -1 && howYChanged === 0) {
    return 'rotate(-0.5turn)';
  }
  if (howXChanged === 0 && howYChanged <= -1) {
    return 'rotate(0.25turn)';
  }
  // ...
}

rotateOrganism = (currentX, newX, currentY, newY, organism) => {
  const icon = organism.icon;
  const howXChanged = currentX - newX;
  const howYChanged = currentY - newY;

  const rotation = this.getOrganismRotation(howXChanged, howYChanged);

  icon.style.transform = rotation;
}
```

mwanago commented on src/styles.css:
```diff
@@ -102,3 +107,69 @@ body {
+  font-weight: bold;
+}
+.new-game-popup-container {
+  position: fixed;
```
Avoid using `position: fixed` and `position: absolute`, because it can complicate your layout a lot.
Centering it on the screen can be achieved with flexbox using `justify-content` and `align-items`.

mwanago commented on src/styles.css:
```diff
@@ -102,3 +107,69 @@ body {
+  font-weight: bold;
+}
+.control-buttons {
+  position: fixed;
```
Avoid using position: fixed and position: absolute, because it can complicate your layout a lot.
Centering it on the screen can be achieved with flexbox using justify-content and align-items.

mwanago commented on src/styles.css:
```diff
@@ -102,3 +107,69 @@ body {
+  height: 170px;
+}
+.control-buttons.active {
+  transform: translate(-50%, -50%) scale(1);
```
Avoid using position: fixed and position: absolute, because it can complicate your layout a lot.
Centering it on the screen can be achieved with flexbox using justify-content and align-items.

MarekKluk committed:
Renamed branch
Used npm prettier on the codebase.
Declared map as a property of the Board class
